### PR TITLE
docs for subscription using apollo-client added (fix #444)

### LINKED ISSUE #444: add example of setting up apollo-client subscriptions
We can use this guide as a starting point or atleast link to it: https://blog.hasura.io/moving-from-apollo-boost-to-graphql-subscriptions-with-apollo-client-cc0373e0adb0

We should add a code snippet to make the usage of the subscription component clear too. 
(Via @jrhicks on discord)
screen shot 2018-09-13 at 11 54 15 pm

The problem is that if forget the "subscription" keyword in the gql-query string (no 3 from the diagram above) it works like a query instead of a subscription. Because this code that sets up apollo-link doesn't work:
```javascript
const link = split(
  // split based on operation type
  ({ query }) => {
    const { kind, operation } = getMainDefinition(query);
    console.log({ query: query, kind: kind, operation: operation });
    return kind === 'OperationDefinition' && operation === 'subscription';
  },
  wsLink,
  httpLink,
)```

Now you'd expect apollo-client to not require declaring the subscription keyword three times, so this is surprising and confusing if you happen to miss it. We should raise an issue at apollo as well.

### PR DESCRIPTION
**Pull request by CodHeK**
### Description

Added a new file subscription.rst containing docs to integrate subscriptions using apollo-client, mentioning the changes that need to made when switching from <Query> to <Subscription>

What component does this PR affect? 

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Docs update
- [ ] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

**Comment by praveenweb** [comment]
@rikinsk - can you review this?

**Comment by CodHeK** [comment]
@rikinsk it was actually to generate the html files that could then be rendered when the python server is started. It was mentioned in the docs, but is actually only needed for development.

**Comment by praveenweb** [comment]
@CodHeK - can you fix the conflicts after merging with master?

**Comment by rikinsk** [comment]
@CodHeK also, check out the docs contributing guidelines and make appropriate fixes

**Comment by CodHeK** [comment]
@praveenweb  @rikinsk  resolved the conflicts, what else should I change ?

File: docs/graphql/manual/guides/integrations/index.rst [hunk 1]
```diff
@@ -3,15 +3,20 @@ Guides: Integration/migration tutorials
...
 Articles:
 ^^^^^^^^^
+
 - `Blog: Move from firebase to realtime GraphQL on Postgres <https://blog.hasura.io/firebase2graphql-moving-from-firebase-to-realtime-graphql-on-postgres-4d36cb7f4eaf>`__.
 - `Blog: Create a Gatsby site using GraphQL on Postgres <https://blog.hasura.io/create-gatsby-sites-using-graphql-on-postgres-603b5dd1e516>`__.
 - `Blog: Instant GraphQL on AWS RDS <https://blog.hasura.io/instant-graphql-on-aws-rds-1edfb85b5985>`__.
 - `Blog: Using TimescaleDB with Hasura GraphQL <https://blog.hasura.io/using-timescaledb-with-hasura-graphql-d05f030c4b10>`__.
 - :doc:`Auth0 JWT Integration with Hasura GraphQL <auth0-jwt>`
 
+Integrations:
```
Comment (rikinsk) [inline docs/graphql/manual/guides/integrations/index.rst]: Remove this section. Add the link under the list above as `` :doc:`subscriptions` ``

File: docs/graphql/manual/guides/integrations/index.rst [hunk 2]
```diff
@@ -3,15 +3,20 @@ Guides: Integration/migration tutorials
...
+
 .. toctree::
   :maxdepth: 1
   :titlesonly:
   :hidden:
-
-  Auth0 JWT Integration <auth0-jwt>
\ No newline at end of file
+  
+  subscriptions
```
Comment (rikinsk) [inline docs/graphql/manual/guides/integrations/index.rst]: rename this file to something more specific, like: apollo-subscriptions

File: docs/graphql/manual/guides/integrations/subscriptions.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
+Subscriptions using apollo-client
```
Comment (rikinsk) [inline docs/graphql/manual/guides/integrations/subscriptions.rst]: Update title to `Setting up GraphQL subscriptions using apollo-client`

File: docs/graphql/manual/guides/integrations/subscriptions.rst [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+=========================================
+
+Setup :
+-------
+Assuming you have the basic apollo-client working as per
+https://www.apollographql.com/docs/react/essentials/get-started.html and now you want to enable subscriptions.
+
+Packages to be installed :
+^^^^^^^^^^^^^^^^^^^^
```
Comment (rikinsk) [inline docs/graphql/manual/guides/integrations/subscriptions.rst]: Fix all underlinings. They should have same length as the headings.

File: docs/graphql/manual/guides/integrations/subscriptions.rst [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  import ApolloClient from "apollo-client";
+  // Setup the network "links"
+  import { WebSocketLink } from 'apollo-link-ws';
+  import { HttpLink } from 'apollo-link-http';
+  import { split } from 'apollo-link';
+  import { getMainDefinition } from 'apollo-utilities';
+  import { InMemoryCache } from 'apollo-cache-inmemory';
+
+
+below these imports initialise you client to fetch subcriptions along with query and mutation.
```
Comment (rikinsk) [inline docs/graphql/manual/guides/integrations/subscriptions.rst]: Fix typo in `subscription` at multiple places

**Comment by CodHeK** [comment]
@rikinsk @praveenweb I have made the required changes, please review the commit.

**Pull request merged**